lib/router/src/lib.rs

```diff
@@ -180,6 +180,10 @@ where
 
         router
     }
+
+    pub fn drain(self) -> IndexMap<Rec::Target, Svc> {
```

Comment:
Nit: I would expect a `drain()` method to return an iterator. I think `get_make` or something similar could serve as a better name for what this is used for.